File: 19. Remove Nth Node From End of List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* dummy = new ListNode(-1);
        dummy->next = head;
        ListNode* pl = dummy, *pf = head;
        for (int i = 0; i < n; i++)
            pf = pf->next;
        while (pf != NULL)
        {
            pl = pl->next;
            pf = pf->next;
        }
        pl->next = pl->next->next;
        return dummy->next;
    }
};

File: 19. Remove Nth Node From End of List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
var removeNthFromEnd = function(head, n) {
    let dummy = new ListNode(-1);
    dummy.next = head;
    let pl = dummy, pf = head;
    for (let i = 0; i < n; i++)
        pf = pf.next;
    while (pf !== null)
    {
        pl = pl.next;
        pf = pf.next;
    }
    pl.next = pl.next.next;
    return dummy.next;
};
